src/uwtools/rocoto.py

```diff
@@ -182,13 +182,47 @@ def _add_task_dependency(self, e: Element, config: dict) -> None:
         :param config: Configuration data for this element.
         """
         e = SubElement(e, STR.dependency)
-        for key, block in config.items():
-            tag, _ = self._tag_name(key)
-            if tag == STR.taskdep:
-                self._set_attrs(SubElement(e, STR.taskdep), block)
+        for tag, block in config.items():
+            if tag in [STR.taskdep, STR.datadep, STR.timedep]:
+                self._add_task_dependency_operand(e, tag=tag, block=block)
+            elif tag in [STR.and_, STR.nand, STR.nor, STR.not_, STR.or_, STR.xor]:
+                self._add_task_dependency_operator(e, config={tag: block})
+            elif tag in [STR.streq, STR.strneq]:
+                self._add_task_dependency_strequality(e, tag=tag, block=block)
+            else:
+                raise UWConfigError("Unhandled dependency type %s" % tag)
+
+    def _add_task_dependency_operand(self, e: Element, tag: str, block: dict) -> None:
+        """
+        ???
+        """
+        self._set_attrs(SubElement(e, tag), block)
+
+    def _add_task_dependency_operator(self, e: Element, config: dict) -> None:
+        """
+        ???
+        """
+        operands = (STR.datadep, STR.taskdep, STR.timedep)
+        operators = (STR.and_, STR.nand, STR.nor, STR.not_, STR.or_, STR.xor)
+        strequality = (STR.streq, STR.strneq)
+
+        for tag, block in config.items():
+            if tag in operators:
+                e = SubElement(e, tag)
```

Comment:
Here is that call to add a subelement to `e`.

Reply:
Is an assignment ok here? It looks to me like `SubElement()` returns the newly-created element so, in a `for` loop like this, wouldn't these calls create a chain of elements, rather than attaching multiple child elements to a single parent? I mean, in pseudocode, something like
```
e is E1
e = SubElement(e, tag2)
e is now E2, and we have E1 -> E2
e = SubElement(e, tag3)
e is now E3, and we have E1 -> E2 -> E3
etc.
```
Without the assignment, the identify of `e` wouldn't change and the loop would just add (potentially) multiple children to `e`.

Should this statement perhaps relocate to `_add_task_dependency_operator()`? Other methods' (`_add_metatask()`, `_add_task()`, `_add_task_dependency()`, etc.) bodies start with this statement, to create the child element and to redefine `e` _for the duration of the method call_ for subsequent processing.

Reply:
I think here I included within the for loop because of how the looping works. We could start the method with a call to add each of these elements to redine `e` and that would require passing the tag along to the method as we do at the beginning of the method call for others as you pointed out. Then ultimately we would discard the rest of the usage for `tag` since for recursion we only care about the `block`.

Reply:
OK, I understand now why you have the `SubElement()` call here, in the body of the loop, but I think the assignment is a bug, which maybe our test data would only exercise if it contained operators nested deeply enough. I think we could maybe just
```
self._add_task_dependency_operator(SubElement(e, tag), config=block)
```
instead of having the `e = SubElement(e, tag)` statement. That would have the effect of 1. Adding the child element, and 2. Passing that child element down in the recursive call, but would not change the identity of `e` within the `for` loop, which is what I think we wouldn't want to do. What do you think?

Also, do we need to pass `dependency_constants` as an argument? I think maybe not -- all methods should be able to access `self._dependency_constants` the same way `_add_task_dependency()` does.

Reply:
Updated this in the newest push.